### ISSUE TITLE
train loss = nan

### ISSUE THREAD

**Issue by BonbonKim** [AIRC-KETI/ke-t5#8]
안녕하세요.
좋은 사전 학습 모델 공개해주셔서 감사드립니다. 연구에 큰 도움이 됩니다.

같은 데이터, 코드로 다른 pretrained t5를 이용하여 finetuning 하면 제대로 동작하고, 
ke-t5-large를 이용하여 finetuning하면 train loss값으로 nan이 나오는 문제가 있습니다..
ke-t5-large의 checkpoint가 잘못 업로드 되어있는지, 제가 하이퍼파라미터를 잘못 주었는지.. 문의드립니다.

저는 허깅페이스 허브에서 pre-trained model(KETI-AIR/ke-t5-large) 파일을 다운로드하여, 
모델은 pytorch_model.bin, config.json, 토크나이저는 spiece.model, tokenizer_config.json를 사용해서 text generation task로 finetuning 진행 중입니다.

실험에 사용한 하이퍼 파라미터는 다음과 같습니다.
batch_size=32, lr=5e-5, warmup_ratio=0, dropout=0.1, gpus=0,1,2, accelerator=ddp, num_best=-1,  ...

model은 T5ForConditionalGeneration을 상속받아 사용중이며, 
model의 vocab size는 64,128입니다. (from config.json)

tokenizer는 다음과 같이 사용중입니다.
import sentencepiece
tokenizer = sentencepiece.SentencePieceProcessor()
tokenizer.Load("./spiece.model") (from config.json)
여기에 13개의 special token을 추가하여, tokenizer의 최종 vocab size는 64,013입니다.

아래는 finetuning 시, loss=nan 확인한 log입니다.
Epoch 0:   0%|          | 0/1785 [00:00<?, ?it/s] [W reducer.cpp:1258] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
[W reducer.cpp:1258] Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration,  which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())
/home/ubuntu/.local/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:131: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
/home/ubuntu/.local/lib/python3.8/site-packages/torch/optim/lr_scheduler.py:131: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. "
Epoch 0:  15%|█▌        | 275/1785 [05:03<27:44,  1.10s/it, loss=**nan**, v_num=41]

**Comment by BonbonKim** [comment]
제가 precision=16을 사용하였는데, 이 때문에 생긴 문제였습니다.
precision=32로 수정하면 제대로 finetuning됩니다.

https://github.com/Lightning-AI/lightning/issues/12600
